api/catalog/api/controllers/search_controller.py

```diff
@@ -425,10 +424,9 @@ def _get_result_and_page_count(
     :return: Result and page count.
     """
     result_count = response_obj.hits.total.value
-    natural_page_count = int(result_count / page_size)
-    if natural_page_count % page_size != 0:
-        natural_page_count += 1
-    page_count = min(natural_page_count, MAX_TOTAL_PAGE_COUNT)
```

Comment:
This clamping is moved up into the paginator and out of the search logic.

Reply:
So glad to see this function simplified even more!